### ISSUE TITLE
New Rule S1763: All code should be reachable

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#7402]
S1763

Some statements (return, break, continue, goto, switch) and throw expressions move control flow out of the current code block. So any unlabeled statements that come after such a jump are unreachable, and either this dead code should be removed, or the logic should be corrected.

```csharp
int fun(int a) {
  int i = 10;
  return i + a;       // Noncompliant
  i++;             // dead code
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1740](https://sonarsource.atlassian.net/browse/NET-1740)

[NET-1740]: https://sonarsource.atlassian.net/browse/NET-1740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
